### PR TITLE
Sparks component ✨

### PR DESCRIPTION
**Pull request by coderitual**
sparks

https://user-images.githubusercontent.com/8572321/220912887-a21dfaaa-2c62-4b03-95f0-98720ebce158.mp4

https://user-images.githubusercontent.com/8572321/220912907-e1306516-96fe-44aa-a01e-1e6872866c48.mp4

https://user-images.githubusercontent.com/8572321/220912916-931653f0-3774-499f-b47e-e7a8b5c1e3d2.mp4

CleanShot 2023-03-01 at 11 11 45

### CODE REVIEW

File: src/components/sparks/Particle.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+
+export interface ParticleCSSProperties extends React.CSSProperties {
+  '--particle-dir'?: number;
+  '--particle-offset'?: number;
+  '--index'?: number;
```
Comment (clxandstuff) [inline src/components/sparks/Particle.tsx:6]: What do you think about prefixing it like other vars `--particle-index` ?

BTW I can't see the usage of this var,  let's remove it then.

File: src/components/sparks/Particle.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          viewBox="0 0 512 512"
+        >
+          {shapes[shape]}
+        </svg>
+      </div>
+    );
+  }
+);
+
+Particle.displayName = 'Particle';
```
Comment (clxandstuff) [inline src/components/sparks/Particle.tsx:49]: I think we can remove it due `@babel/plugin-transform-react-display-name` is doing that for us. We have it as a part of [@babel/preset-react](https://babeljs.io/docs/babel-preset-react)

File: src/components/sparks/Sparks.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      <Sparks>
+        <button type="button">Click me</button>
+      </Sparks>
+    );
+
+    expect(sparks.container.firstChild).toBeTruthy();
+    expect(sparks.getByRole('button', {name: /click me/i})).toBeTruthy();
+  });
+
+  it('renders around interactive element and this element still captures all the events', () => {
```
Comment (clxandstuff) [inline src/components/sparks/Sparks.spec.tsx:30]: I would rephrase it a little bit 'renders around interactive element and this element still can be clicked' because we actually testing it but not all events.

File: src/components/sparks/Sparks.tsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+              />
+            )
+          )}
+        </div>
+      </div>
+    );
+  }
+);
+
+Sparks.displayName = 'Sparks';
```
Comment (clxandstuff) [inline src/components/sparks/Sparks.tsx:175]: I think we can remove it due @babel/plugin-transform-react-display-name is doing that for us. We have it as a part of [@babel/preset-react](https://babeljs.io/docs/babel-preset-react)

File: src/components/sparks/Sparks.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+describe('<Sparks />', () => {
+  it('renders correctly and displays its children', () => {
+    const sparks = render(
+      <Sparks>
+        <button type="button">Click me</button>
+      </Sparks>
+    );
+
+    expect(sparks.container.firstChild).toBeTruthy();
```
Comment (clxandstuff) [inline src/components/sparks/Sparks.spec.tsx:26]: If you add role on svg ('img') , you can use getByRole and check more accurately sparks.

**Review by clxandstuff — CHANGES_REQUESTED** [review]

File: src/components/sparks/Particle.tsx (hunk 1 above)
Comment (coderitual) [reply src/components/sparks/Particle.tsx:6]: Yes it is leftover, thx for catching

File: src/components/sparks/Sparks.tsx (hunk 4 above)
Comment (coderitual) [reply src/components/sparks/Sparks.tsx:184]: True

File: src/components/sparks/Sparks.spec.tsx (hunk 5 above)
Comment (coderitual) [reply src/components/sparks/Sparks.spec.tsx:26]: Yea, honestly the meaning of this test is vague (checks if element just renders anything)

I will remove it once I figure out chromatic tests

File: src/components/sparks/Particle.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+
+export interface ParticleCSSProperties extends React.CSSProperties {
+  '--particle-dir'?: number;
```
[the remark this answers is not in the document]
Comment (coderitual) [reply src/components/sparks/Particle.tsx:4]: It controls direction of the particle - up or down represented  by 1/-1. Thanks to that direction is controlled entirely within key frame (—particle-dir * <translateinpx>). Some particles are going from top to bottom and some the opposite

File: .eslintrc.rules.js [hunk 7]
```diff
@@ -506,7 +507,7 @@ module.exports = {
     // require method and property shorthand syntax for object literals
     'object-shorthand': ['error', 'always'],
     // suggest using of const declaration for variables that are never modified after declared
-    'prefer-const': 'error',
```
[the remark this answers is not in the document]
Comment (coderitual) [reply .eslintrc.rules.js:509]: Temp, I want to discuss with you next week

File: src/components/sparks/Sparks.spec.tsx (hunk 5 above)
Comment (coderitual) [reply src/components/sparks/Sparks.spec.tsx:26]: As for role img it is pointless here since all contaniner is hidden in accesibility tree. The only thing I care about is that component renders. Expect is just used to satisfy test.

prpbably this test can be skipped because existence of 2 other tests, it was just the only test for some time.

Comment (clxandstuff) [reply src/components/sparks/Sparks.spec.tsx]: That's the best case if any other tests cover this we can get rid of this artificial "it renders" tests

File: src/components/sparks/Particle.tsx (hunk 2 above)
Comment (clxandstuff) [reply src/components/sparks/Particle.tsx:48]: You missed it or should we keep it here? I see that in Sparks same part has been removed.

Comment (coderitual) [reply src/components/sparks/Particle.tsx]: Sorry, missed it, fixed now

**Review by clxandstuff — APPROVED** [review]
